app/src/main/java/io/intrepid/contest/rest/RetrofitClient.java

```diff
@@ -1,43 +1,64 @@
 package io.intrepid.contest.rest;
 
+import android.support.annotation.NonNull;
 import android.support.annotation.VisibleForTesting;
 
+import com.google.gson.FieldNamingPolicy;
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+import com.jakewharton.retrofit2.adapter.rxjava2.RxJava2CallAdapterFactory;
+
 import java.util.concurrent.TimeUnit;
 
 import io.intrepid.contest.BuildConfig;
+import io.intrepid.contest.settings.UserSettings;
 import okhttp3.OkHttpClient;
+import okhttp3.Request;
 import okhttp3.logging.HttpLoggingInterceptor;
+import retrofit2.Converter;
 import retrofit2.Retrofit;
 import retrofit2.converter.gson.GsonConverterFactory;
 import timber.log.Timber;
 
 public class RetrofitClient {
 
-    // TODO: change this to a real endpoint
-    private static final String BASE_URL = "https://api.ipify.org/";
+    public static final String ACCEPT_APPLICATION = "vnd.judgy-server.herokuapp.com";
+    private static final String BASE_URL = "http://judgy-server.herokuapp.com/api/";
     private static final int CONNECTION_TIMEOUT = 30;
+    private static final int API_VERSION = 1;
+    private static RestApi restApi;
 
-    private static RestApi instance;
+    private RetrofitClient() {
+    }
 
-    public static RestApi getApi() {
-        if (instance == null) {
-            instance = createRestApi(BASE_URL);
-        }
-        return instance;
+    public static void init(UserSettings userSettings) {
+        restApi = createRestApi(BASE_URL, userSettings);
     }
 
-    @VisibleForTesting
-    static RestApi getTestApi(String baseUrl) {
-        return createRestApi(baseUrl);
+    public static RestApi getApi() {
+        return restApi;
     }
 
-    private RetrofitClient() {
+    @VisibleForTesting
+    static RestApi getTestApi(String baseUrl, @NonNull UserSettings userSettings) {
+        return createRestApi(baseUrl, userSettings);
     }
 
-    private static RestApi createRestApi(String baseUrl) {
-        OkHttpClient.Builder builder = new OkHttpClient.Builder();
+    private static RestApi createRestApi(String baseUrl, UserSettings userSettings) {
+        OkHttpClient.Builder builder = new OkHttpClient.Builder()
+                .addInterceptor(chain -> {
+                    Request request = chain.request();
```

Comment:
Actually I don't think I can, because in the end I need to use `chain.proceed(newRequest)`... I'm not quite sure if there's another way around it.